fdbserver/EncryptKeyProxyInterface.h

```diff
@@ -90,66 +90,67 @@ struct HaltEncryptKeyProxyRequest {
 	}
 };
 
+struct EKPBaseCipherDetails {
+	constexpr static FileIdentifier file_identifier = 2149615;
+	int64_t encryptDomainId;
+	uint64_t baseCipherId;
+	StringRef baseCipherKey;
+
+	EKPBaseCipherDetails() : encryptDomainId(0), baseCipherId(0), baseCipherKey(StringRef()) {}
+	explicit EKPBaseCipherDetails(int64_t dId, uint64_t id, StringRef key, Arena& arena)
+	  : encryptDomainId(dId), baseCipherId(id), baseCipherKey(StringRef(arena, key)) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, encryptDomainId, baseCipherId, baseCipherKey);
+	}
+};
+
 struct EKPGetBaseCipherKeysByIdsReply {
 	constexpr static FileIdentifier file_identifier = 9485259;
 	Arena arena;
-	std::unordered_map<uint64_t, StringRef> baseCipherMap;
+	std::vector<EKPBaseCipherDetails> baseCipherDetails;
 	int numHits;
 	Optional<Error> error;
 
 	EKPGetBaseCipherKeysByIdsReply() : numHits(0) {}
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		serializer(ar, arena, baseCipherMap, numHits, error);
+		serializer(ar, arena, baseCipherDetails, numHits, error);
 	}
 };
 
 struct EKPGetBaseCipherKeysByIdsRequest {
 	constexpr static FileIdentifier file_identifier = 4930263;
 	UID requesterID;
-	std::vector<uint64_t> baseCipherIds;
+	std::unordered_map<uint64_t, int64_t> baseCipherIdMap;
```

Comment:
shall this be `std::vector<pair<int64_t, uint64_t>>`? This interface should not assume cipher ids being unique among domains.

Reply:
Good point. 
I started with std::vector and then realized it is better to dedup the _ids_ on the client side to avoid extra server work, however, it is possible for KMS _not_ to maintain unique KeyId semantics across _domainIds_. 

Shall update the code.

Reply:
or maybe `std::unordered_set<pair<int64_t, uint64_t>>`. not sure if it is serializable.

Reply:
unordered_set serialization isn't working the last time I tried it. My plan for now is to add code to dedup using ordered_set and convert to std::vector, and maybe debug unordered_set RPC serialization separately (add a TODO in the code for now)

Reply:
Done. 
Thanks